decompressors.go

```diff
@@ -0,0 +1,50 @@
+package fakemachine
+
+import (
+	"io"
+	"compress/gzip"
+
+	// BSD 3-clause licenses
+	"github.com/klauspost/compress/zstd"
+	"github.com/ulikunitz/xz"
+)
+
+func ZstdDecompressor(dst io.Writer, src io.Reader) error {
+	decompressor, err := zstd.NewReader(src)
+	if err != nil {
+		return err
+	}
+	defer decompressor.Close()
+
+	_, err = io.Copy(dst, decompressor)
+	return err
+}
+
+func XzDecompressor(dst io.Writer, src io.Reader) error {
+	decompressor, err := xz.NewReader(src)
+	if err != nil {
+		return err
+	}
+	// There is no Close() API just yet, see
+	// https://github.com/ulikunitz/xz/issues/45
+	//defer decompressor.Close()
```

Comment:
seems like in the linked issue the GC takes care of closing things - so we can drop these three lines ?

Reply:
It captures nicely why the xz path is asymmetrical, while providing useful upstream reference for details. Do you feel so strongly against it?

Reply:
well i think the comment assumes that the API will be added in future - but the author explicitly says on the linked issue there is no need - perhaps rewording that would be better to avoid misleading ourselves in the future ? :-)

how about something like:
```
// There is no Close() API due to Go garbage collection cleaning up the object. See: <link>
```

Reply:
Ack updated.